frontend/public/components/environment.jsx

```diff
@@ -103,16 +103,46 @@ export class EnvironmentPage extends PromiseComponent {
    * Reset the page to initial state
    * @private
    */
-  _clearChanges() {
+  _reload() {
     const {rawEnvData} = this.props;
     this.setState({
       currentEnvVars: envVarsToArray(rawEnvData),
       errorMessage: null,
       success: null,
-      modified: false
+      modified: false,
+      stale: false
     });
   }
 
+  /**
+   * Build out our currentEnvVars state object from our incoming props.
+   * If there is a change and are read/write let the user know we have updated vars otherwise just refresh the page.
+   * For no change return null
+   *
+   * @param nextProps
+   * @param prevState
+   */
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { currentEnvVars } = prevState;
+    const { rawEnvData, readOnly } = nextProps;
+    const incomingEnvVars = envVarsToArray(rawEnvData);
+    if (!_.isEqual(incomingEnvVars, currentEnvVars)) {
```

Comment:
I'd suggest an early return to avoid nested `if`s. You also shouldn't need to call `envVarsToArray` twice. I might use a ternary operator as well.

```js
if (_.isEqual(incomingEnvVars, currentEnvVars)) {
  return null;
}

return readOnly ? { currentEnvVars: incomingEnvVars } : { stale: true, success: null };
```